bitcoin/src/pow.rs

```diff
@@ -385,13 +387,13 @@ define_extension_trait! {
             }
             // Comments relate to the `pow.cpp` file from Core.
             // ref: <https://github.com/bitcoin/bitcoin/blob/0503cbea9aab47ec0a87d34611e5453158727169/src/pow.cpp>
-            let min_timespan = params.pow_target_timespan >> 2; // Lines 56/57
-            let max_timespan = params.pow_target_timespan << 2; // Lines 58/59
+            let min_timespan = (params.pow_target_timespan >> 2) as i64; // Lines 56/57
+            let max_timespan = (params.pow_target_timespan << 2) as i64; // Lines 58/59
```

Comment:
In 0503cbea9aab47ec0a87d34611e5453158727169:

I think we should just change these constants to be `i64`s to avoid all these casts.

Reply:
@shinghim not if the values we need to compare it to are all signed. Then making it unsigned requires a bunch of casts.

Reply:
@apoelstra that makes sense, though when i changed that const to an `i64`, it let to me having to cast in a couple of places:
```rust
 pub fn difficulty_adjustment_interval(&self) -> u64 {
        self.pow_target_timespan as u64 / self.pow_target_spacing
 }
```

and 

```rust
// Part of from_next_work_required, line 397
let retarget = retarget.div((params.pow_target_timespan as u128).into());
```

I don't see any issues in deviating from the core implementation and using an `i64` for `pow_target_timespan`, but seeing as how changing it will require adding casts elsewhere, what are your thoughts on keeping it as a `u64` to more closely align with core?

Reply:
Can we make it a `u32` so that we can use `.into` to convert it to either a u64 or an i64?

If not, then pick the one with the fewest required casts :P.